tests/programs.test.ts

```diff
@@ -1,57 +1,77 @@
 import test from 'tape'
-import { readFileSync } from 'node:fs'
 
 import { promiseRunner, charlieStashSeed, setupTest } from './testing-utils'
 import { addProgram } from '../src/flows/programs/add'
 import { viewPrograms } from '../src/flows/programs/view'
 import { removeProgram } from '../src/flows/programs/remove'
-import { AddProgramParams } from '../src/flows/programs/types'
+import { deployProgram } from '../src/flows/programs/deploy'
 
 const networkType = 'two-nodes'
 
 test('programs', async t => {
   const { run, entropy } = await setupTest(t, { seed: charlieStashSeed, networkType })
-  await run('charlie stash register', entropy.register())
-  const noopProgram: any = readFileSync(
-    new URL('./programs/program_noop.wasm', import.meta.url)
-  )
-  const newPointer = await run(
-    'deploy',
-    entropy.programs.dev.deploy(noopProgram)
-  )
-
-  const noopProgramInstance: AddProgramParams = {
-    programPointer: newPointer,
-    programConfig: '',
-  }
-
-  t.test('Add Program', async ap => {
-    const runAp = promiseRunner(ap)
-
-    const programsBeforeAdd = await runAp('get programs initial', entropy.programs.get(entropy.programs.verifyingKey))
-    ap.equal(programsBeforeAdd.length, 1, 'charlie has 1 program')
-    await runAp('adding program', addProgram(entropy, noopProgramInstance))
-    const programsAfterAdd = await runAp('get programs after add', entropy.programs.get(entropy.programs.verifyingKey))
-    ap.equal(programsAfterAdd.length, 2, 'charlie has 2 programs')
-    ap.end()
+  await run('register', entropy.register()) // TODO: consider removing this in favour of just testing add
+
+  let programPointer1
+
+  t.test('programs - deploy', async t => {
+    const run = promiseRunner(t)
+
+    programPointer1 = await run (
+      'deploy!',
+      deployProgram(entropy, {
+        bytecodePath: './tests/programs/program_noop.wasm'
+      })
+    )
+
+    t.end()
   })
 
-  t.test('Remove Program', async rp => {
-    const runRp = promiseRunner(rp)
-    const programsBeforeRemove = await runRp('get programs initial', entropy.programs.get(entropy.programs.verifyingKey))
-  
-    rp.equal(programsBeforeRemove.length, 2, 'charlie has 2 programs')
-    await runRp('removing noop program', removeProgram(entropy, { programPointer: newPointer, verifyingKey: entropy.programs.verifyingKey }))
-    const programsAfterRemove = await runRp('get programs initial', entropy.programs.get(entropy.programs.verifyingKey))
-    rp.equal(programsAfterRemove.length, 1, 'charlie has 1 less program')
-    rp.end()
+  const getPrograms = () => entropy.programs.get(entropy.programs.verifyingKey)
```

Comment:
Can use this instead of calling entropy directly if you want
```suggestion
  const getPrograms = () => viewPrograms(entropy, { verifyingKey: entropy.programs.verifyingKey })
```